mlrun/api/api/endpoints/tags.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import fastapi
+import fastapi.concurrency
+import sqlalchemy.orm
+
+import mlrun.api.api.deps
+import mlrun.api.crud.tags
+import mlrun.api.schemas
+import mlrun.api.utils.auth.verifier
+import mlrun.api.utils.singletons.project_member
+
+router = fastapi.APIRouter()
+
+
+@router.post("/projects/{project}/tags/{tag}")
+async def overwrite_object_tags_with_tag(
+    project: str,
+    tag: str,
+    objects: mlrun.api.schemas.TagsObjects,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: sqlalchemy.orm.Session = fastapi.Depends(
+        mlrun.api.api.deps.get_db_session
+    ),
+):
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    for object_list in objects.objects:
+        # check permission per object type
+        await fastapi.concurrency.run_in_threadpool(
+            mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions,
+            getattr(mlrun.api.schemas.AuthorizationResourceTypes, object_list.kind),
+            project,
+            resource_name=None,
+            # not actually overwriting objects, just overwriting the objects tags
+            action=mlrun.api.schemas.AuthorizationAction.update,
+            auth_info=auth_info,
+        )
+
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.crud.Tags().overwrite_object_tags_with_tag,
+        db_session,
+        project,
+        tag,
+        objects.objects,
+    )
+    return {}
```

Comment:
Usually `POST`/`PUT` should return the resource created. In this case, since we're in the `tags` endpoint, this should contain the tag "object" - which in our case should simply be the name of the tag + the project.
Please add to the schema an object which describes this return value, for example `Tag` and add to the function decorator: `response_model=schemas.Tag`, see feature-store endpoints for an example.